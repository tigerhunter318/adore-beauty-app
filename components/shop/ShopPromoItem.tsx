import React from 'react'
import Container from '../ui/Container'
import Type from '../ui/Type'
import ResponsiveImage from '../ui/ResponsiveImage'
import theme from '../../constants/theme'
import { formatPromotionData } from '../promotions/utils/helpers'
import ImageSize from '../../constants/ImageSize'

const styleSheet = {
  carouselItem: {
    height: 325,
    backgroundColor: theme.white,
    shadowColor: theme.black,
    shadowOffset: { width: 2, height: 1 },
    shadowOpacity: 0.1,
    shadowRadius: 5,
    marginBottom: 10,
    paddingBottom: 10
  },
  itemImageWrapper: {
    width: '100%',
    backgroundColor: theme.white,
    marginTop: 5,
    height: 202
  },
  itemImageContainer: {
    width: 130,
    height: 130
  },
  title: {
    height: 44
  },
  description: {
    lineHeight: 24,
    letterSpacing: 0.5,
    color: theme.lightBlack
  }
}

const colors = [theme.darkPeach, theme.darkOrange]

type ShopPromoItemsProps = {
  item: {}
  index: number
  onPromoPress: (item: any) => void
}

const ShopPromoItem = ({ item, index, onPromoPress }: ShopPromoItemsProps) => {
  const { code, isAddFreeItem, description, title, image } = formatPromotionData(item)
  const hasCouponCode = !isAddFreeItem && code
  const { width, height } = ImageSize.product.medium
  const handlePressPromo = () => onPromoPress({ item })

  return (
    <Container
      testID="ShopPromoItem"
      key={index}
      align
      ph={0.5}
      style={[
        styleSheet.carouselItem,
        {
          backgroundColor: colors[index % colors.length]
        }
      ]}
      activeOpacity={1}
      onPress={handlePressPromo}
    >
      <Container style={styleSheet.itemImageWrapper} center justify="space-between">
        <Container style={styleSheet.title} align mt={1.5}>
          <Type heading size={14} center ph={2.2} lineHeight={22} bold numberOfLines={2}>
            {title}
          </Type>
        </Container>
        <Container style={styleSheet.itemImageContainer} mt={1.3}>
          {image && <ResponsiveImage src={image} width={width} height={height} useAspectRatio />}
        </Container>
      </Container>
      <Container pt={1}>
        {hasCouponCode && (
          <Type bold center ph={1.6} numberOfLines={2}>
            USE PROMO CODE{' '}
            <Type bold center color={theme.white}>
              {' '}
              {code}
            </Type>
          </Type>
        )}
        <Type
          numberOfLines={hasCouponCode ? 3 : 4}
          pt={0.5}
          size={13}
          left
          ph={2.1}
          letterSpacing={0.09}
          lineHeight={20}
          center
        >
          {description}
        </Type>
      </Container>
    </Container>
  )
}

export default ShopPromoItem
