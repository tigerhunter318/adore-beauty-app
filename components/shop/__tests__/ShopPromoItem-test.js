import React from 'react'
import renderer from 'react-test-renderer'
import ShopPromoItem from '../ShopPromoItem'

const mockData = require('./__mocks__/ShopPromoItem.json')

it(`ShopPromoItem renders content`, () => {
  const tree = renderer.create(<ShopPromoItem item={mockData} onPromoPress={jest.fn()} index={0} />).toJSON()

  expect(tree).toMatchSnapshot()
})
