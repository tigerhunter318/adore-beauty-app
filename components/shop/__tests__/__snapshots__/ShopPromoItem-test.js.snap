// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopPromoItem renders content 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E3BDB1",
      "height": 325,
      "marginBottom": 10,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 5,
      "paddingRight": 5,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 2,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 5,
    }
  }
  testID="ShopPromoItem"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "space-between",
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "height": 202,
          "marginTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginTop": 15,
          },
          Object {
            "height": 44,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        ph={2.2}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Bold",
              "fontSize": 14,
              "lineHeight": 22,
              "paddingLeft": 22,
              "paddingRight": 22,
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            undefined,
          ]
        }
      >
        Choose your Beauty Bonus Gift!
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 13,
          },
          Object {
            "height": 130,
            "width": 130,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "paddingTop": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      allowFontScaling={false}
      numberOfLines={2}
      ph={1.6}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-Bold",
            "paddingLeft": 16,
            "paddingRight": 16,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      USE PROMO CODE
       
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Bold",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
         
        BONUS
      </Text>
    </Text>
    <Text
      allowFontScaling={false}
      numberOfLines={3}
      ph={2.1}
      pt={0.5}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-Regular",
            "fontSize": 13,
            "letterSpacing": 0.09,
            "lineHeight": 20,
            "paddingLeft": 21,
            "paddingRight": 21,
            "paddingTop": 5,
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
      Jane Iredale Lash Duo Gift With Purchase
    </Text>
  </View>
</View>
`;
